backend/pong/game_logic.py

```diff
@@ -77,37 +79,44 @@ def check_collisions(self):
 
     def handle_player_collision(self, player_y):
         # Calculate new angle based on collision with player
-        relative_intersect_y = (self.player1_y + PLAYER_HEIGHT / 2) - self.ball_y
+        relative_intersect_y = (self.left_player_y + PLAYER_HEIGHT / 2) - self.ball_y
         normalized_intersect_y = relative_intersect_y / (PLAYER_HEIGHT / 2)
         bounce_angle = normalized_intersect_y * (MAX_BOUNCE_ANGLE - MIN_BOUNCE_ANGLE)
         self.ball_angle = 180 - bounce_angle
 
-    def handle_wall_collision(self, wall):
-        # Game over, set winner and loser
-        self.game_over = True
+    def handle_wall_collision(self, wall):       
         if wall == 'left':
-            self.winner = 'player2'
-            self.loser = 'player1'
+            self.right_player_score += 1
         elif wall == 'right':
-            self.winner = 'player1'
-            self.loser = 'player2'
+            self.left_player_score += 1
+            
+        if self.left_player_score == 5:
+            self.game_over = True
+            self.winner = 'left_player'
+            self.loser = 'right_player'
+        elif self.right_player_score == 5:
+            self.game_over = True
+            self.winner = 'right_player'
+            self.loser = 'left_player'
+
 
     def get_game_state(self):
-        # Construct JSON message with player and ball positions
         game_state = {
```

Comment:
I somehow overlooked this part in the previous review 😅 Would maybe be good to have a bit more structure too the json eg:

 game_state = {
            'ball': {
                'position': {
                    'x': self.ball_x,
                    'y': self.ball_y
                },
                'size': BALL_WIDTH,   # I assume the ball is round/square
            },
            'players': {
                  'size': {.   #unless we go for games customisation, the sizes will be the same, so no reason to duplicate it
                      'width': PLAYER_WIDTH,
                      'height': PLAYER_HEIGHT
                  },
                'left': {
                    'position': {
                        'x': self.left_player_x,
                        'y': self.left_player_y
                    },
                    'score': self.left_player_score
                },
                'right': {
                    'position': {
                        'x': self.right_player_x,
                        'y': self.right_player_y
                    },
                    'score': self.right_player_score
                }
            },
            'game': {
                'wall_margin': WALL_MARGIN,
                'over': self.game_over,
                'winner': self.winner,
                'loser': self.loser
            }
        }
        self.send(text_data=json.dumps(game_state))

Reply:
Sorry the indentation vanished...

Reply:
Oh true! Would probably make sense to make a struct / object with these values and parse into a json object and back!

Reply:
would make it easier to also then inject the ai instead of another player!